cached_proc_macro/src/lib.rs

```diff
@@ -162,21 +192,31 @@ pub fn cached(args: TokenStream, input: TokenStream) -> TokenStream {
         _ => panic!("cache types (unbound, size, time, or type and create) are mutually exclusive"),
     };
 
-    // make the set cache block
-    let set_cache_block = match (&args.result, &args.option) {
-        (false, false) => quote! { cache.cache_set(key, result.clone()); },
-        (true, false) => quote! {
-            match result.clone() {
-                Ok(result) => { cache.cache_set(key, Ok(result)); },
-                _ => {},
-            }
-        },
-        (false, true) => quote! {
-            match result.clone() {
-                Some(result) => { cache.cache_set(key, Some(result)); },
-                _ => {},
-            }
-        },
+    // make the set cache and return cache blocks
+    let (set_cache_block, return_cache_block) = match (&args.result, &args.option) {
+        (false, false) => {
+            let set_cache_block = quote! { cache.cache_set(key, result.clone()); };
+            let return_cache_block = quote! { return result.clone(); };
```

Comment:
I know you just moved this line up from below, but I think this might have an (existing) unintentional and unnecessary clone that we can remove

Reply:
Can you expand on your thinking here? Because first we need to clone the value because it moves into `cache_set()`, and we still need to be able to return it to our caller. And then when we `cache_get` we only get a reference, so we need to clone since while the cache is static, its entries do not have a guaranteed lifetime, right?

Reply:
oh.. yes. Sorry I thought this was applying as a return statement for all cases, which it is not!